gw_eccentricity/eccDefinition.py

```diff
@@ -251,24 +297,178 @@ def __init__(self, dataDict, num_orbits_to_exclude_before_merger=2,
+            "amplm_zeroecc",    # Dict of amplitude of quasicircular waveform modes
+            "phaselm_zeroecc",  # Dict of phase of quasicircular waveform modes
+            "omegalm_zeroecc",  # Dict of omega of quasicircular waveform modes
         ]
         return list_of_keys
 
-    def truncate_dataDict_if_necessary(self,
-                                       dataDict,
-                                       num_orbits_to_exclude_before_merger,
-                                       extra_kwargs):
-        """Truncate dataDict if "num_orbits_to_exclude_before_merger" is not None.
+    def get_amp_phase_omega_data(self, dataDict):
+        """Get the dictionary of amplitude, omega and phase from user dataDict.
+
+        The `dataDict` provided by the user can contain any of the data
+        corresponding to the keys in `get_recognized_dataDict_keys`. To compute
+        eccentricity and mean anomaly, we need amplitude, phase and omega. In
+        this function, we get these data from `dataDict` and create a new
+        dictionary that contains only the amplitude, phase and omega of the
+        available modes.
+
+        For example, if `dataDict` has only the complex hlm modes, this
+        functions uses the hlm modes to create a dictionary containing the
+        amplitude, phase and omega by decomposing the hlm modes. Afterwards,
+        only these waveform data in the new dataDict is used for all the
+        further computations.
+        """
+        amp_phase_omega_dict = {}
+        # add amplm and amplm_zeroecc if zeroecc data provided
+        amp_phase_omega_dict.update(self.get_amplm_from_dataDict(dataDict))
+        # add phaselm and phaselm_zeroecc if zeroecc data provided
+        amp_phase_omega_dict.update(self.get_phaselm_from_dataDict(dataDict))
+        # add omegalm
+        if "omegalm" in dataDict:
+            amp_phase_omega_dict.update({"omegalm": dataDict["omegalm"]})
+        else:
+            # compute it from phaselm that is already in amp_phase_omega_dict
+            amp_phase_omega_dict.update(
+                {"omegalm": self.get_omegalm_from_phaselm(
+                    dataDict["t"],
+                    amp_phase_omega_dict["phaselm"])})
+
+        # add omegalm_zeroecc if zeroecc data is provided
+        if "omegalm_zeroecc" in dataDict:
+            amp_phase_omega_dict.update(
+                {"omegalm_zeroecc": dataDict["omegalm_zeroecc"]})
+        # Look for zeroecc phaselm is amp_phase_omega_dict and compute
+        # omegalm_zeroecc from it if phaselm_zeroecc is present
+        elif "phaselm_zeroecc" in amp_phase_omega_dict \
+             and "t_zeroecc" in dataDict:
+            amp_phase_omega_dict.update(
+                {"omegalm_zeroecc": self.get_omegalm_from_phaselm(
+                    dataDict["t_zeroecc"],
+                    amp_phase_omega_dict["phaselm_zeroecc"])})
+        else:
+            pass
```

Comment:
Why do you need this else condition if you're just going to do pass?

Reply:
my bad. I assumed that I need to end the if-else always with an else, but apparently, it can just end with an elif. Removed it now.